### PR TITLE
Add `Module widget concept` Concret Code Exemple

### PR DESCRIPTION
**Pull request by PululuK**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Add `Module widget concept` Concret Code Exemple
| Fixed ticket? | N/A

### CODE REVIEW

**Review by NeOMakinG — APPROVED** [review]

File: src/content/1.7/modules/concepts/widgets.md [hunk 1]
```diff
@@ -104,4 +104,80 @@ The function `renderWidget()` of a specific module can be called directly:
 Hook::coreRenderWidget(Module $module, $hook_name, $params);
 ```
 
+## Code exemple
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/widgets.md:107]:
```suggestion
## Code example
```

File: src/content/1.7/modules/concepts/widgets.md [hunk 2]
```diff
@@ -104,4 +104,80 @@ The function `renderWidget()` of a specific module can be called directly:
...
+```php
+<?php
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+use PrestaShop\PrestaShop\Core\Module\WidgetInterface;
+
+
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/widgets.md:121]:
```suggestion

```

File: src/content/1.7/modules/concepts/widgets.md [hunk 3]
```diff
@@ -104,4 +104,80 @@ The function `renderWidget()` of a specific module can be called directly:
...
+    }
+
+    public function renderWidget($hookName = null, array $configuration = []) 
+    {
+        $this->smarty->assign($this->getWidgetVariables($hookName, $configuration));
+
+        return $this->display(__FILE__, 'views/templates/widget/mymodule.tpl');
+    }
+ 
+    public function getWidgetVariables($hookName = null , array $configuration = [] )
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/widgets.md:152]:
```suggestion
    public function getWidgetVariables($hookName = null , array $configuration = [])
```

File: src/content/1.7/modules/concepts/widgets.md [hunk 4]
```diff
@@ -104,4 +104,80 @@ The function `renderWidget()` of a specific module can be called directly:
...
+        
+        return [
+            'my_var1' => 'my_var1_value',
+            'my_var2' => 'my_var2_value',
+            'my_var_n' => 'my_var_n_value',
+            'my_dynamic_var_by_param' => $this->getMyDynamicVarByParamKey($myParamKey),
+        ];
+    }
+    
+    public function getMyDynamicVarByParamKey($paramKey)
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/widgets.md:164]:
```suggestion
    public function getMyDynamicVarByParamKey(string $paramKey)
```

File: src/content/1.7/modules/concepts/widgets.md [hunk 5]
```diff
@@ -104,4 +104,80 @@ The function `renderWidget()` of a specific module can be called directly:
...
+
+        parent::__construct();
+
+        $this->displayName = $this->l('My module');
+        $this->description = $this->l('Description of my module.');
+
+        $this->confirmUninstall = $this->l('Are you sure you want to uninstall?');
+    }
+
+    public function renderWidget($hookName = null, array $configuration = [])
```
Comment (Progi1984) [inline src/content/1.7/modules/concepts/widgets.md:145]:
```suggestion
    public function renderWidget($hookName, array $configuration) 
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: src/content/1.7/modules/concepts/widgets.md [hunk 6]
```diff
@@ -104,4 +104,80 @@ The function `renderWidget()` of a specific module can be called directly:
...
+        $this->description = $this->l('Description of my module.');
+
+        $this->confirmUninstall = $this->l('Are you sure you want to uninstall?');
+    }
+
+    public function renderWidget($hookName = null, array $configuration = []) 
+    {
+        $this->smarty->assign($this->getWidgetVariables($hookName, $configuration));
+
+        return $this->display(__FILE__, 'views/templates/widget/mymodule.tpl');
```
[the remark this answers is not in the document]
Comment (Progi1984) [reply src/content/1.7/modules/concepts/widgets.md:147]: @PululuK : @kpodemski has a good suggestion.

[the remark this answers is not in the document]
Comment (PululuK) [reply src/content/1.7/modules/concepts/widgets.md:147]: I found nothing in the doc about that

[the remark this answers is not in the document]
Comment (kpodemski) [reply src/content/1.7/modules/concepts/widgets.md:147]: @PululuK yes, but as you can see all, or almost all, native modules are using `fetch`, https://github.com/PrestaShop/PrestaShop/pull/6491

[the remark this answers is not in the document]
Comment (PululuK) [reply src/content/1.7/modules/concepts/widgets.md]: Thanks @kpodemski  !
Bisous 😘

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @PululuK

**Pull request merged**